fbw-common/src/wasm/terronnd/src/navigationdisplay/display.h

```diff
@@ -113,14 +114,37 @@ class Display : public DisplayBase {
     this->_frameData->defineArea(side == DisplaySide::Left ? FrameDataLeftName : FrameDataRightName);
     this->_frameData->requestArea(SIMCONNECT_CLIENT_DATA_PERIOD_ON_SET);
     this->_frameData->setOnChangeCallback([=]() {
-      this->destroyImage();
-
       if (!this->_ignoreNextFrame && this->_configuration.terrainActive) {
-        this->_nanovgImage =
-            nvgCreateImageMem(this->_context, 0, this->_frameData->data().data(), static_cast<int>(this->_frameBufferSize));
         if (this->_nanovgImage == 0) {
-          std::cerr << "TERR ON ND: Unable to decode the image from the stream" << std::endl;
+          // If we don't have an image yet, create one
+          this->_nanovgImage = nvgCreateImageMem(this->_context, 0, this->_frameData->data().data(), static_cast<int>(this->_frameBufferSize));
+          if (this->_nanovgImage == 0) {
+            fprintf(stderr, "TERR ON ND: Unable to decode the image from the stream. Reason: %s\n", stbi_failure_reason());
```

Comment:
Any reason not to use a stream for printing the error like before?

Reply:
That's because streams add a new line on every `<<` in MSFS which makes the log hard to read so this seems like the easiest way to get around that.

Reply:
We have the fmt lib now available. Check it out.

Reply:
> That's because streams add a new line on every `<<` in MSFS which makes the log hard to read so this seems like the easiest way to get around that.

You can use this as IU did in the cpp framework:
fbw-common/src/wasm/cpp-msfs-framework/lib/fmt
Example:
fbw-common/src/wasm/cpp-msfs-framework/lib/logging.h